### ISSUE TITLE
3.0.2209 and mingw

### ISSUE THREAD

**Issue by 3rav** [ICLDisco/parsec#460]
With 3.0.2209 version I have this error:

```
[127/129] Building C object parsec/data_dist/matrix/CMakeFiles/parsec-data-matrix.dir/apply.c.obj
[128/129] Building C object parsec/data_dist/matrix/CMakeFiles/parsec-data-matrix.dir/diag_band_to_rect.c.obj
[129/129] Linking Fortran shared library parsec\libparsec.dll
FAILED: parsec/libparsec.dll parsec/libparsec.dll.a
cmd.exe /C "cd . && C:\msys64\mingw64\bin\gfortran.exe -O3 -DNDEBUG -O3  -pipe -shared -o parsec\libparsec.dll -Wl,--out-implib,parsec\libparsec.dll.a -Wl,--major-image-version,3,--minor-image-version,0 @CMakeFiles\parsec.rsp  && cd ."
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: parsec/fortran/CMakeFiles/parsec_fortran.dir/parsecf.c.obj:parsecf.c:(.text+0x74): undefined reference to `strsep'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: parsec/fortran/CMakeFiles/parsec_fortran.dir/parsecf.c.obj:parsecf.c:(.text+0xef): undefined reference to `strsep'
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Comment by devreal** [comment]
@3rav it looks like the use of strsep was removed sometime after the release. Could you try whether current master compiles on MinGW? We're close to releasing 4.0.

**Comment by 3rav** [comment]
''
C:/msys64/usr/local/pkg_parsec/src/parsec-git/parsec/utils/output.c:113:22: note: while refere

ncing 'info'

  113 | static output_desc_t info[PARSEC_OUTPUT_MAX_STREAMS];

      |                      ^~~~

In function 'open_file',

    inlined from 'output.isra' at C:/msys64/usr/local/pkg_parsec/src/parsec-git/parsec/utils/o

utput.c:931:39:

C:/msys64/usr/local/pkg_parsec/src/parsec-git/parsec/utils/output.c:705:9: warning: 'strncpy'

specified bound 260 equals destination size [-Wstringop-truncation]

  705 |         strncpy(filename, output_dir, MAXPATHLEN);

      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

[37/175] [BISON][parsec_yacc] Building parser with bison 3.7.3

FAILED: parsec/interfaces/ptg/ptg-compiler/parsec.y.c parsec/interfaces/ptg/ptg-compiler/parse

c.y.h C:/msys64/usr/local/pkg_parsec/src/build-x86_64-w64-mingw32-static/parsec/interfaces/ptg

/ptg-compiler/parsec.y.c C:/msys64/usr/local/pkg_parsec/src/build-x86_64-w64-mingw32-static/pa

rsec/interfaces/ptg/ptg-compiler/parsec.y.h

cmd.exe /C "cd /D C:\msys64\usr\local\pkg_parsec\src\build-x86_64-w64-mingw32-static\parsec\in

terfaces\ptg\ptg-compiler && C:\msys64\mingw64\bin\bison.exe -d -o C:/msys64/usr/local/pkg_par

sec/src/build-x86_64-w64-mingw32-static/parsec/interfaces/ptg/ptg-compiler/parsec.y.c C:/msys6

4/usr/local/pkg_parsec/src/parsec-git/parsec/interfaces/ptg/ptg-compiler/parsec.y"

bison.exe: /mingw64/share/bison/m4sugar/m4sugar.m4: cannot open: No such file or directory

[38/175] [FLEX][parsec_flex] Building scanner with flex 2.6.4

ninja: build stopped: subcommand failed.

''

**Comment by devreal** [comment]
The output about `strncpy` is only a warning. The bison issue seems to be a problem with MinGW, as far as I can tell from this mailing list entry: https://mingw-users.narkive.com/ZlXMOjFq/bison-3-0-2-and-m4sugar-m4

I don't know much about MinGW or the changes made since PaRSEC 3.0 was released. Did we change the use of bison since then @bosilca? 

Anyway, having a working bison installation is crucial, i.e., this command should not fail:
```
C:\msys64\usr\local\pkg_parsec\src\build-x86_64-w64-mingw32-static\parsec\interfaces\ptg\ptg-compiler && C:\msys64\mingw64\bin\bison.exe -d -o C:/msys64/usr/local/pkg_parsec/src/build-x86_64-w64-mingw32-static/parsec/interfaces/ptg/ptg-compiler/parsec.y.c C:/msys64/usr/local/pkg_parsec/src/parsec-git/parsec/interfaces/ptg/ptg-compiler/parsec.y
```

**Comment by bosilca** [comment]
It turned out to be difficult to find a recent enough version installed on most target systems, so we decided to ship our own generated (lex and yacc) files, and use them if the local lex/yacc version was too old. So, there is no real depepdency on lex/yacc anymore.

Also, we replaced `strsep` with the POSIX-compliant `strtok`, which should exists everywhere (including MSYS and MINGW).

Btw, I am able to compile the current version on my windows box, so I think the real issue here is the one @devreal pointed out (issue with your local bison).
